Feature: Course Instance Redesign

GeneralDido opened the pull request:
### Event show view (app/views/events/show.html.erb)

  - Remove the collapsing sidebar so the show page runs single-column.
  - Restructure the action row: show a SciLifeLab-branded “Register” CTA when a registration URL exists, restyle “View event”, keep identifiers/star, and move the calendar button into the summary card.
  - Delegate the body to a composable details layout, followed by the Twitter card, report panel, and moderation tools.

  ———

  ### Presenter helpers (app/helpers/events_helper.rb)

We centralise markup and formatting, suppress empty sections, and keep the view partials declarative by buidling the following helpers:
  - `event_section_card`
  - `event_detail_item`
  - `event_pill_list`
  - `event_cost_value`
  - `event_formatted_datetime` 

  ———

  ### Details layout refactor (app/views/events/partials/details/…)

  - Split each card into its own partial 
      - `description_card` 
      - `learning_card`
      - `prereq_card`
      - `topics_card`
      - `affiliation_card`
      - `audience_card`
      - `recognition_card`
      - `materials_card`
      - `external_resources_card`
      - `history_card`
      - `submission_card`
      - `summary_card`
      - `content_providers_card`

  - The coordinating `_details_sections.html.erb` now just renders those partials, pairs them where needed, and manages the intro two-column layout vs. stacked flow.

  ———

  ### SCSS modularisation (app/assets/stylesheets/events/…)

  - We build focused scss partials:
    - `_actions.scss` for CTA button themes.
    - `_layout.scss` for card grids, intro columns, and spacing.
    - `_typography.scss` for shared headings and rich-text sizing.
    - `_detail_items.scss` for the label/value grid and calendar CTA wrapper.
    - `_pills.scss` for chip styling and colour variants.
    - `_content_providers.scss` for the provider card internals.
    - `_header.scss` for show-page title/icon sizing.
    - `_responsive.scss` for all breakpoint adjustments.
  - `events.scss` now just imports those modules plus the legacy map styles, keeping related rules grouped by concern.

  ———

  ### Brand palette (app/assets/stylesheets/themes/traininghub.scss:7)

  - Register SciLifeLab teal, grape, lime, aqua, and grey tones (including 25/50/75 tints) as `$scilifelab-color-*` variables. The new button and provider-card styles reference these constants so future components can reuse the same palette.

  ———

  ### Net effect

  - Actions are clearer, with registration/canonical links styled to match brand guidance.
  - Event metadata is delivered through focused cards using shared presenter helpers; there’s no monolithic template to parse.
  - Description vs. summary columns stay aligned and pair cards collapse gracefully on small screens.
  - Content providers and other sections inherit consistent typography and pill styling.
  - All changes are limited to views and SCSS; no controller/model behaviour altered.

 
<img width="958" height="1652" alt="image" src="https://github.com/user-attachments/assets/89987039-ec6c-44bc-a573-06f7f178afff" />

GeneralDido committed:
initial design update

- update helpers
- build partials and update show.html.erb
update stylesheets/events.scss

GeneralDido committed:
make it similar to the mock-up design

GeneralDido committed:
optimisations

GeneralDido committed:
remove sidebar code

GeneralDido committed:
make content providers card not stretch till the end of Description card

GeneralDido committed:
fix prereq__heading, make title and logo nicer

GeneralDido committed:
add colors from scilifelab brand

GeneralDido committed:
remove unused css

GeneralDido committed:
Merge remote-tracking branch 'origin/rc-1.4.0' into feat/course_instance_redesign

GeneralDido committed:
refactor

GeneralDido committed:
update cost basis to match previous functionality

adeel-ahmad-awan reviewed (approved):
lgtm

kristen-schroeder commented:
@GeneralDido a few changes requested:

Fix 1 - circular cropping in content provider card does not work for rectangular logos
Fix 2 - only some content provider logos are transparent, remove green background from card so those with a white background display properly

see:
<img width="255" height="662" alt="Screenshot 2025-10-02 at 16 09 26" src="https://github.com/user-attachments/assets/6f4fe431-68f4-4529-89f8-2b007b575226" />

Fix 3 - change 'view event' button to read 'view course website' to be consistent with course announcement form.
Fix 4 - add little arrow to 'register' button to denote it sending you to an external page as well

see:
<img width="257" height="61" alt="Screenshot 2025-10-02 at 16 17 54" src="https://github.com/user-attachments/assets/d2154816-a7f5-42d1-901c-ef3dbd81277d" />

GeneralDido committed:
update with fixes for Kristen

GeneralDido commented:
@kristen-schroeder you can check again and if it's ok approve so @harshita3199 can take a look :)

kristen-schroeder committed:
suggested tweaks to external link button and icon size

kristen-schroeder reviewed (changes requested):
@GeneralDido see commit [74bfb65](https://github.com/SciLifeLab-TrainingHub-Platform/TeSS/pull/336/commits/74bfb65d616189e3c91a3ab9a4b20e868f929fc3) - suggesting to use a black icon on the Register button (hacky solution, pls update to something more elegant if needed) and to resize the provider icons based on width instead of height.

GeneralDido committed:
Refine external link button icon handling and provider logo sizing

Unified external_link_button to accept a customizable icon, updated the event registration button to use the dark arrow, and reinstated an explicit height cap on content provider
  logos to prevent layout stretch.

GeneralDido committed:
Merge branch 'rc-1.4.0' into feat/course_instance_redesign

GeneralDido commented:
@kristen-schroeder more changes made based on your commit. Please review + approve if good.

kristen-schroeder reviewed (approved):
lgtm!

harshita3199 commented:
After testing I found out few things, mentioned in this [ticket](https://app.zenhub.com/workspaces/th-developmentdesigning-board-6565ea6f792dae0625b5c739/issues/gh/scilifelab-traininghub-platform/tess/293)

harshita3199 reviewed (changes requested):
just the contact info changes as it hides out. see this [ticket](https://app.zenhub.com/workspaces/th-developmentdesigning-board-6565ea6f792dae0625b5c739/issues/gh/scilifelab-traininghub-platform/tess/293)

GeneralDido committed:
Merge branch 'rc-1.4.0' into feat/course_instance_redesign

GeneralDido committed:
fix overflow in event_detail_item

harshita3199 reviewed (approved):
Lgtm
